mod-kitpvp/src/main/java/gg/solarmc/loader/kitpvp/KitPvp.java

```diff
@@ -122,6 +123,92 @@ public StatisticResult addAssists(Transaction transaction, int amount) {
         return new StatisticResult(newValue);
     }
 
+    /**
+     * Adds experience to the user account. Infallible.
+     * @param transaction represents the transaction
+     * @param amount represents the amount of xp to add
+     * @return a result with the new value of xp
+     */
+    public StatisticResult addExperience(Transaction transaction, int amount) {
+        if (amount <= 0) {
+            throw new IllegalArgumentException("amount must be positive");
+        }
+        KitpvpStatisticsRecord statisticsRecord = getStatistics(transaction);
+
+        int existingValue = statisticsRecord.getExperience();
+        int newValue = existingValue + amount;
+
+        statisticsRecord.setExperience(newValue);
+        statisticsRecord.store(KITPVP_STATISTICS.EXPERIENCE);
+
+        this.updateExperience(newValue);
+
+        return new StatisticResult(newValue);
+    }
+
+    /**
+     * Adds current killstreaks to the user account. Infallible.
+     * @param transaction represents the current killstreak
+     * @param amount represents the amount of killstreak to add
+     * @return a result with the new value of killstreak
+     */
+    public StatisticResult addCurrentKillstreaks(Transaction transaction, int amount) {
+        if (amount <= 0) {
+            throw new IllegalArgumentException("amount must be positive");
+        }
+        KitpvpStatisticsRecord statisticsRecord = getStatistics(transaction);
+
+        int existingValue = statisticsRecord.getCurrentKillstreak();
+        int newValue = existingValue + amount;
+
+        statisticsRecord.setCurrentKillstreak(newValue);
+        statisticsRecord.store(KITPVP_STATISTICS.CURRENT_KILLSTREAK);
+
+        this.updateCurrentKillstreak(newValue);
+
+        return new StatisticResult(newValue);
+    }
+
+    /**
+     * Sets the killstreak amouont. Infallible
+     * @param transaction represents the current transaction
+     * @param amount represents the amount of killstreak to set to
+     */
+    public void setCurrentKillstreaks(Transaction transaction, int amount) {
+        if (amount <= 0) {
+            throw new IllegalArgumentException("amount must be positive");
+        }
+        KitpvpStatisticsRecord statisticsRecord = getStatistics(transaction);
+
+        statisticsRecord.setCurrentKillstreak(amount);
+        statisticsRecord.store(KITPVP_STATISTICS.CURRENT_KILLSTREAK);
+
+        this.updateCurrentKillstreak(amount);
+    }
+
+    /**
+     * Adds highest killstreaks to the user account. Infallible.
+     * @param transaction represents the current transaction
+     * @param amount represents the amount of killstreak to add
+     * @return a result with the new value of killstreak
+     */
+    public StatisticResult addHighestKillstreaks(Transaction transaction, int amount) {
```

Comment:
What purpose is served by incrementing the highest killstreak? (I suggest looking at my next comment before addressing this one)

Reply:
already addressed above